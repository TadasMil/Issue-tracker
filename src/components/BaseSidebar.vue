<script setup lang="ts" name="BaseSidebar">
import type { IPage } from '@/types'
import { useRoute } from 'vue-router'

const pages: IPage[] = [
  { name: 'Home', path: '/', icon: 'bi bi-house-door' },
  { name: 'Open Issues', path: '/issues/open', icon: 'bi bi-folder2-open' },
  {
    name: 'Completed Issues',
    path: '/issues/completed',
    icon: 'bi bi-folder-check',
  },
  { name: 'Trashed Issues', path: '/issues/trashed', icon: 'bi bi-folder-x' },
]

const route = useRoute()

const isActive = (page: IPage) => route.path === page.path
</script>
<template>
  <header class="h-100">
    <b-navbar :container="false" variant="dark" class="h-100 py-0">
      <b-navbar-nav align class="d-flex h-100 flex-column h-100">
        <b-nav-item
          v-for="page in pages"
          :key="page.path"
          :to="page.path"
          :active="isActive(page)"
        >
          <i :class="`${page.icon} text-light fa-2x p-2`"></i>
        </b-nav-item>
      </b-navbar-nav>
    </b-navbar>
  </header>
</template>

<style #scoped>
.active {
  background-color: var(--bs-secondary);
}
</style>
