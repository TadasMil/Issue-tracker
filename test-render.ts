import type { RenderOptions } from '@testing-library/vue'
import BootstrapVue3 from 'bootstrap-vue-3'
import { render } from '@testing-library/vue'

const customRender = (ui: any, options?: RenderOptions) =>
  render(ui, {
    ...options,
    global: {
      ...options?.global,
      plugins: [BootstrapVue3, ...(options?.global?.plugins || [])],
    },
  })

export * from '@testing-library/vue'
export { customRender as render }
