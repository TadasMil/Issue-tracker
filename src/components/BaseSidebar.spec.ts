import { describe, it, expect, vi } from 'vitest'
import { render } from '../../test-render'

import BaseSidebar from './BaseSidebar.vue'

vi.mock('vue-router', () => ({
  useRoute: vi.fn(() => ({ path: '/' })),
}))

describe('BaseSidebar', () => {
  it('should render the component given home page is active', () => {
    const { html } = render(BaseSidebar)

    expect(html()).toMatchSnapshot()
  })
})
