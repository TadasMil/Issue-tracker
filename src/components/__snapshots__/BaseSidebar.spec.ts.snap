// Vitest Snapshot v1

exports[`BaseSidebar > should render the component given home page is active 1`] = `
"<header class=\\"h-100\\">
  <nav class=\\"navbar bg-dark navbar-expand h-100 py-0\\">
    <ul class=\\"navbar-nav justify-content- d-flex h-100 flex-column h-100\\">
      <li class=\\"nav-item\\"><a class=\\"active nav-link\\" to=\\"/\\" href=\\"/\\" target=\\"_self\\"><i class=\\"bi bi-house-door text-light fa-2x p-2\\"></i></a></li>
      <li class=\\"nav-item\\"><a class=\\"nav-link\\" to=\\"/issues/open\\" href=\\"/issues/open\\" target=\\"_self\\"><i class=\\"bi bi-folder2-open text-light fa-2x p-2\\"></i></a></li>
      <li class=\\"nav-item\\"><a class=\\"nav-link\\" to=\\"/issues/completed\\" href=\\"/issues/completed\\" target=\\"_self\\"><i class=\\"bi bi-folder-check text-light fa-2x p-2\\"></i></a></li>
      <li class=\\"nav-item\\"><a class=\\"nav-link\\" to=\\"/issues/trashed\\" href=\\"/issues/trashed\\" target=\\"_self\\"><i class=\\"bi bi-folder-x text-light fa-2x p-2\\"></i></a></li>
    </ul>
  </nav>
</header>"
`;
